docs/design/assetgeneration.md

```diff
@@ -55,15 +57,48 @@ type File struct {
 }
 ```
 
+## Target Asset
+
+A target asset is an asset that generates files to write to disk, and is also able to be loaded from disk.
+The typical target assets are `install-config`, `manifests`, `ignition configs`, `cluster`, etc.
+After being loaded and consumed by a children asset, the existing on-disk target asset will be purged.
+E.g.
+
+```shell
+$ openshift-install install-config
+# Generate install-config.yml
+
+$ openshift-install manifests
+# Generate manifests/ and tectonic/ dir, also remove install-config.yml
+```
+
+A target asset looks like this:
+
+```go
+type TargetAsset interface{
```

Comment:
Do we have a reason why we need a new interface `TargetAsset`.
It seems like Load should be part of the WritableAsset.

If we want a separate interface, rename to some thing that conveys `Loadable`. An would like to see explicit examples of why we need this interface.

This name `Target*` is very overloaded and often semantically misused.